homework_02 - внесение изменений и дополнений.

HenryBuckinson opened the pull request:
Task №1:
1) Проверка на null элемента внутри массива RAW_DATA, а также, если поле name объекта Person == null.
2) Поправил вывод по Вашему требованию, но реализовал две версии метода для формирования карты. Если принципиально важно, чтобы в качестве значения карты был список - оставлю список, а если число - оставлю число.

Task№2:
1) Оптимизировал алгоритм до О(N)
2) Обработал исключения в главном методе программы. 

Task№3:
1) Оптимизировал алгоритм до O(N)

HenryBuckinson committed:
homework_02 - внесение изменений и дополнений.

HenryBuckinson committed:
Merge branch 'master' into developer

Lada303 commented on homework_02/src/main/java/secondTask/SecondTaskRunner.java:
```diff
@@ -1,25 +1,60 @@
+     */
+    public static int[] coupleOfSumNumber(int[] targetArray, int resultSumNumber) throws ArrayDataException, ArraySizeException, ArithmeticException {
+        if (targetArray == null) {
+            throw new ArrayDataException("На входной массив отсутствует ссылка");
```
ЗАЧЕМ ВЫКИДЫВАТЬ ИСКЛЮЧЕНИЯ!!!

HenryBuckinson committed:
homework_02, Task №2:
1) Убрал выброс исключений.
2) Теперь, в случае, если входные данные некорректны, или нет подходящей пары чисел в качестве результата возвращается пустой массив.
